src/libraries/Common/src/System/Net/Http/NoWriteNoSeekStreamContent.cs

```diff
@@ -29,9 +29,10 @@ internal NoWriteNoSeekStreamContent(Stream content)
         protected override Task SerializeToStreamAsync(Stream stream, TransportContext context) =>
             SerializeToStreamAsync(stream, context, CancellationToken.None);
 
+#if HTTP_DLL || !NETSTANDARD
```

Comment:
We shouldn't need the "HTTP_DLL" define anymore. It was used only when we were building WinHttpHandler as part of System.Net.Http.dll. But @ManickaP PR got rid of building WinHttpHandler as part of System.Net.Http.dll.

So, you should remove "HTTP_DLL" define from this file and any other place it might appear such as CSPROJ files, etc.

Reply:
As far as I can tell WinHttpHandler and System.Net.Http link to NoWriteNoSeekStreamContent.cs. 

I've been fighting with the build for the past couple hours regarding this as it seems the netfx version of WinHttpHandler targets a different definition of HttpContent and therefore requires a different signature for the changed method (SerializeToStreamAsync).

Reply:
>I've been fighting with the build for the past couple hours regarding this as it seems the netfx version of WinHttpHandler targets a different definition of HttpContent and therefore requires a different signature for the changed method (SerializeToStreamAsync).

Yes, WinHttpHandler doesn't build against 'netcoreapp'. It builds against 'netstandard20' which won't have the new APIs.

You might need to add new defines to work around this problem.

Reply:
I think I get it, but please correct me @davidsh if I'm wrong. The HTTP_DLL should be removed, but very similar constant must be introduced for WinHttpHandler library with pretty much the opposite meaning to HTTP_DLL. Then filter overrides and access modifier according to this.

Reply:
Correct.

We are now introducing new APIs in .NET Core which affects the building of the WinHttpHandler library (the new overloads with cancellationtoken). So we need to use compile-time mechanisms to preserve the ability to build WinHttpHandler against 'netstandard20' targets (which include .NET Framework). And 'netstandard20' target doesn't have those new APIs available.

We have two different ways to achieve compile-time differences. One is thru the use of these defines. The other is partial classes and separating out into different files (with suffixes like .netcoreapp.cs or .netfx.cs etc.) like we did for other Http stuff.  But when the code is scattered like this it is easier to use defines similar to 'HTTP_DLL'.

Reply:
Considering only this one class is problematic in this case (with a single 'real' method), would it make sense to duplicate the file for System.Net.Http and the WinHttpHandler?

Reply:
>would it make sense to duplicate the file for System.Net.Http and the WinHttpHandler

No. We don't like to duplicate any sources. That's why we put files into Common folders etc.

For small things like this, it is easier to just use compile-time define constants.